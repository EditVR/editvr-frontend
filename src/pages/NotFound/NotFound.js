/**
 * @file NotFound.js
 * Exports a React component that render's EditVR's NotFound route.
 */

import React from 'react';
import PropTypes from 'prop-types';
import { Typography, withStyles } from '@material-ui/core';

import { ThemeProvider } from '../../hoc';
import NotFoundStyles from './NotFound.style';

const NotFound = ({ classes: { wrapper, text } }) => (
  <div className={wrapper}>
    <Typography className={text} variant="headline">
      Not Found
    </Typography>
    <Typography className={text}>
      The page you requested does not exist.
    </Typography>
  </div>
);

NotFound.propTypes = {
  classes: PropTypes.shape({
    wrapper: PropTypes.string.isRequired,
    text: PropTypes.string.isRequired
  }).isRequired
};

export default ThemeProvider(withStyles(NotFoundStyles)(NotFound));
