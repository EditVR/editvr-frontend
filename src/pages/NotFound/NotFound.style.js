/**
 * @file NotFound.style.js
 * Exports NotFound component styles.
 */

export default theme => ({
  wrapper: {
    background: theme.palette.background.default,
    height: '100vh',
    maxHeight: '100%'
  },
  text: {
    paddingTop: 200,
    textAlign: 'center'
  }
});
