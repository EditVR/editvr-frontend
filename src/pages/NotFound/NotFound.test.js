/**
 * @file NotFound.test.js
 * Contains tests for NotFound.js.
 */

import React from 'react';
import renderer from 'react-test-renderer';

import NotFound from './NotFound';

describe('<NotFound />', () => {
  it('Matches its snapshot', () => {
    expect(renderer.create(<NotFound />).toJSON()).toMatchSnapshot();
  });
});
