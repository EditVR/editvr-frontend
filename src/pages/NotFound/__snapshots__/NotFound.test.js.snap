// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound /> Matches its snapshot 1`] = `
<div
  className="NotFound-wrapper-1"
>
  <h1
    className="MuiTypography-root-3 MuiTypography-headline-8 NotFound-text-2"
  >
    Not Found
  </h1>
  <p
    className="MuiTypography-root-3 MuiTypography-body1-12 NotFound-text-2"
  >
    The page you requested does not exist.
  </p>
</div>
`;
